hs_collection_resource/tests/test_collection.py

```diff
@@ -0,0 +1,349 @@
+        self.assertEqual(resp_json["current_sharing_status"], "Private")
+        self.assertEqual(resp_json["new_sharing_status"], "")
+        self.assertEqual(resp_json["metadata_status"], "Sufficient to make public")
+        self.assertEqual(self.resCollection.metadata.collection.resources.all().count(), 3)
+
+    def test_custom_logic_in_hs_core_1(self):
+
+        # this func tests that the public collection should downgrade to private if its any member resource is changed to private
+
+        self.api_client.logout()
+        # user 2 login
+        self.api_client.login(username='user2', password='mypassword2')
+
+        ##### test custom logic for collection in hs_core
+        # check gen res 5 current sharing status --> private
+        self.assertEqual(self.resGen5_user2.raccess.public, False)
+        self.assertEqual(self.resGen5_user2.raccess.discoverable, False)
+
+        # make gen res 5 public (not working because it does not meet public precondition: no resouece file in generic res)
+        # url_to_set_resource_flag_for_resGen5 = self.url_to_set_resource_flag.format(self.resGen5_user2.short_id)
+        # response = self.api_client.post(url_to_set_resource_flag_for_resGen5, {'t': 'make_public'}, HTTP_REFERER='http://foo/bar')
+
+        # make gen res 5 public manaully
+        self.resGen5_user2.raccess.public = True
+        self.resGen5_user2.raccess.discoverable = True
+        self.resGen5_user2.raccess.save()
+
+        # check gen res 5 current sharing status --> public
+        self.assertEqual(self.resGen5_user2.raccess.public, True)
+        self.assertEqual(self.resGen5_user2.raccess.discoverable, True)
+
+        # check collection current sharing status --> private
+        self.assertEqual(self.resCollection2_user2.raccess.public, False)
+        self.assertEqual(self.resCollection2_user2.raccess.discoverable, False)
+
+        # add one public gen res 5into collection
+        url_to_update_collection_2 = self.url_to_update_collection.format(self.resCollection2_user2.short_id)
+        response = self.api_client.post(url_to_update_collection_2, {'resource_id_list': [self.resGen5_user2.short_id]})
+        resp_json = json.loads(response.content)
+        self.assertEqual(resp_json["status"], "success")
+        self.assertEqual(resp_json["metadata_status"], "Sufficient to make public")
+        self.assertEqual(self.resCollection2_user2.metadata.collection.resources.all().count(), 1)
+
+        # manually make collection public
+        self.resCollection2_user2.raccess.public = True
+        self.resCollection2_user2.raccess.discoverable = True
+        self.resCollection2_user2.raccess.save()
+
+        # check collection current sharing status --> public
+        self.assertEqual(self.resCollection2_user2.raccess.public, True)
+        self.assertEqual(self.resCollection2_user2.raccess.discoverable, True)
+
+        f=open("abcde2.txt", 'w', 0)
```

Comment:
@pkdash thanks for pointing this out, actually I made some changes to this branch to address your comments and it is not done yet. I just pushed current intermediate version before leaving office. I will continue working on it. thanks